File: src/test/scala/photos/brooklyn/practice/codingexercises/scala/leetcode/solutions/OverlappingSquareMatricesTest.scala
package photos.brooklyn.practice.codingexercises.scala.leetcode.solutions

import org.scalatest.{FlatSpec, Matchers}

class OverlappingSquareMatricesTest extends FlatSpec with Matchers {

  behavior of "OverlappingSquareMatrices"

  it should "come up with a 3 in the test case" in {
    val A = Array(Array(1, 1, 0), Array(0, 1, 0), Array(0, 1, 0))
    val B = Array(Array(0, 0, 0), Array(0, 1, 1), Array(0, 0, 1))
    OverlappingSquareMatrices.largestOverlap(A, B) shouldEqual 3
  }

}

File: test/java/photos/brooklyn/practice/codingexercises/java/leetcode/OverlappingSquareMatricesTest.java
package photos.brooklyn.practice.codingexercises.java.leetcode;

import org.junit.Test;

import static org.junit.Assert.*;

public class OverlappingSquareMatricesTest {

    @Test
    public void largestOverlap() {
        final int[][] A = {{1,1,0}, {0,1,0},{0,1,0}};
        final int[][] B = {{0,0,0}, {0,1,1},{0,0,1}};
        assertEquals(3, new OverlappingSquareMatrices().largestOverlap(A, B));
    }

    @Test
    public void largestOverlap2x2() {
        final int[][] A = {{0,1}, {1,1}};
        final int[][] B = {{1,1}, {1,0}};
        assertEquals(3, new OverlappingSquareMatrices().largestOverlap(A, B));
    }
}